build.gradle

```diff
@@ -349,14 +349,26 @@ subprojects {
         }
     }
 
+
     if (!Boolean.getBoolean("skipCheckstyle")) {
         checkstyle {
-            configFile new File(rootDir, "style/checkstyle-rules.xml")
-            configProperties = ["checkstyle.suppressions.file": new File(rootDir, "style/checkstyle-suppressions.xml")]
+            configProperties = [
+                    "checkstyle.suppressions.file": new File(rootDir, "style/checkstyle-suppressions.xml"),
+                    "checkstyle.importcontrol.file": new File(rootDir, "style/import-control.xml")]
             ignoreFailures false
             showViolations true
             toolVersion "${checkstyleVersion}"
         }
+        tasks.withType(Checkstyle) {
+            doFirst {
+                def useJunit5 = project.ext.properties.get("useJunit5", false)
+                if (useJunit5) {
```

Comment:
Could you elaborate on the difference between the two, and why would this be necessary? I am really not a big fan of maintaining two different styling rules. It's confusing for most, especially contributors.

Reply:
The junit5 style includes a error if junit4 classes are used. Is there a better way to make a hard cut like that? There should only be two during the transition period.

Reply:
Ah gotcha. So the plan is to ultimately consolidate down to one, right? That's good with me. Thanks very much!